builder/builder.go

```diff
@@ -80,7 +80,7 @@ type Result struct {
 // ImageCacheBuilder represents a generator for stateful image cache.
 type ImageCacheBuilder interface {
 	// MakeImageCache creates a stateful image cache.
-	MakeImageCache(ctx context.Context, cacheFrom []string) ImageCache
+	MakeImageCache(ctx context.Context, cacheFrom []string) (ImageCache, error)
```

Comment:
Probably would make sense to add this in the first commit as well, as it's fixing up things from that one.

Reply:
Done